### PR TITLE
Add dynamic robots.txt from API

### PR DESCRIPTION
**Pull request by bookernath**
## What/Why?
Use new robots.txt node to dynamically populate robots.txt details from control panel, replacing previous hardcoded approach.

We still dynamically insert the sitemap URL based on the env.

## Testing
Preview deployment

image

### CODE REVIEW

File: core/app/robots.txt/route.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * https://nextjs.org/docs/app/api-reference/file-conventions/metadata/robots
+ *
+ */
+
+import { getChannelIdFromLocale } from '~/channels.config';
+import { client } from '~/client';
+import { graphql } from '~/client/graphql';
+import { defaultLocale } from '~/i18n/routing';
+
+const ROBOTS_TXT_QUERY = graphql(`
```
Comment (chanceaclark) [inline core/app/robots.txt/route.ts:18]:
```suggestion
const RobotsTxtQuery = graphql(`
```

File: core/app/robots.txt/route.ts
Comment (chanceaclark) [inline core/app/robots.txt/route.ts:1]: I'm curious if this file needs to be in `core/app/[locale]` so we can actually do a channel-specific query? I think the `next-intl` routing mechanism will help route the request here 🤔

File: core/app/robots.txt/route.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  process.env.NEXTAUTH_URL || process.env.VERCEL_PROJECT_PRODUCTION_URL || '',
+);
+
+export const GET = async () => {
+  const { data } = await client.fetch({
+    document: ROBOTS_TXT_QUERY,
+    channelId: getChannelIdFromLocale(defaultLocale),
+  });
+
+  let robotsTxtText = data.site.settings?.robotsTxt || '';
```
Comment (chanceaclark) [inline core/app/robots.txt/route.ts:49]: 🍹 Seems redundant to have `TxtText`
```suggestion
  let robotsTxt = data.site.settings?.robotsTxt || '';
```

File: core/app/robots.txt/route.ts
Comment (bookernath) [reply core/app/robots.txt/route.ts:1]: robots.txt must live on the root (in the standard)... maybe I shouldn't even be trying to make the query channel specific?

Comment (chanceaclark) [reply core/app/robots.txt/route.ts:1]: Well we do have channel-specific robot settings so I think we do need to support that. Normally, yeah `robots.txt` should live in the root, but I am curious if next-intl will handle the routing if it's nested under a locale, e.g. `/robots.txt` would route to `/en/robots.txt`? The other channels wouldn't necessarily "work" as no request would be sent to `/da/robots.txt` but it'll enable multi-domain -> single Catalyst instance in the future.

Comment (bookernath) [reply core/app/robots.txt/route.ts:1]: Multi-domain is a great point.

Comment (bookernath) [reply core/app/robots.txt/route.ts:1]: After poking at this, I'm going to recommend we push forward as-is and re-evaluate at the time we add multi-domain support; this ended up being more complex than I expected.

File: core/app/robots.txt/route.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  const { data } = await client.fetch({
+    document: RobotsTxtQuery,
+    channelId: getChannelIdFromLocale(defaultLocale),
+    fetchOptions: { cache: 'no-store' }, // disable caching to get the latest robots.txt at build time
+  });
+
+  let robotsTxt = data.site.settings?.robotsTxt || '';
+
+  // add sitemap to robots.txt
+  robotsTxt += `\nSitemap: ${baseUrl.origin}/sitemap.xml\n`;
```
Comment (chanceaclark) [inline core/app/robots.txt/route.ts:53]: 🍹 
```suggestion
  const robotsTxt = `${data.site.settings?.robotsTxt ?? ''}\nSitemap: ${baseUrl.origin}/sitemap.xml\n`;
```

**Review by chanceaclark — APPROVED** [review]

**Pull request merged**